pkg/lib/config/secret_whatsapp.go

```diff
@@ -106,3 +106,208 @@ func (c *WhatsappOnPremisesCredentials) SensitiveStrings() []string {
 		c.Password,
 	}
 }
+
+// WhatsappCloudAPIAuthenticationTemplateType is NOT an official term.
+// Officially, they just call authentication templates "authentication templates".
+// See https://developers.facebook.com/docs/whatsapp/business-management-api/authentication-templates/
+// The variants are
+//   - One-Tap Autofill Authentication Templates
+//     https://developers.facebook.com/docs/whatsapp/business-management-api/authentication-templates/#one-tap-autofill-authentication-templates
+//   - Copy Code Authentication Templates
+//     https://developers.facebook.com/docs/whatsapp/business-management-api/authentication-templates/#copy-code-authentication-templates
+//   - Zero-Tap Authentication Templates
+//     https://developers.facebook.com/docs/whatsapp/business-management-api/authentication-templates/#zero-tap-authentication-templates
+//
+// We only support the Copy Code variant.
+type WhatsappCloudAPIAuthenticationTemplateType string
+
+const (
+	WhatsappCloudAPIAuthenticationTemplateTypeCopyCodeButton WhatsappCloudAPIAuthenticationTemplateType = "copy_code_button"
+)
+
+type WhatsappCloudAPIAuthenticationTemplateConfig struct {
+	Type           WhatsappCloudAPIAuthenticationTemplateType                  `json:"type"`
+	CopyCodeButton *WhatsappCloudAPIAuthenticationTemplateCopyCodeButtonConfig `json:"copy_code_button"`
+}
+
+type WhatsappCloudAPIAuthenticationTemplateCopyCodeButtonConfig struct {
+	Name      string   `json:"name"`
+	Languages []string `json:"languages"`
+}
+
+type WhatsappCloudAPICredentials struct {
+	PhoneNumberID                string                                        `json:"phone_number_id"`
+	AccessToken                  string                                        `json:"access_token"`
+	AuthenticationTemplateConfig *WhatsappCloudAPIAuthenticationTemplateConfig `json:"authentication_template"`
+}
+
+func (c *WhatsappCloudAPICredentials) SensitiveStrings() []string {
+	return []string{
+		c.PhoneNumberID,
+		c.AccessToken,
+	}
+}
+
+// https://developers.facebook.com/docs/whatsapp/business-management-api/message-templates/supported-languages
+var _ = SecretConfigSchema.Add("WhatsappCloudAPITemplateLanguage", `
```

Comment:
Isn't this list same as the on-prem template languages?

Reply:
Yes, you are correct. They are the same. So we are going to update the list of supported languages in On-Premises API.